### ISSUE TITLE
"securedrop-admin setup" fails with confusing errors

### ISSUE THREAD

**Issue by b-meson** [freedomofpress/securedrop#3200]
# Bug

## Description

We previously had a virtualenv in `~/Persistence/securedrop/.venv` that had the packages needed to admin SecureDrop. In 0.6 we moved the location to `~/Persistence/securedrop/admin/.venv`. Right now, if you delete the _old_ path and run `./securedrop-admin setup -v`, the script errors out with an error. 

## Steps to Reproduce

Note: I was on a 0.5.2 release before I checked out the 0.6 code.

I removed the old virtualenv in `~/Persistence/securedrop/.venv` and attempted to run `./securedrop-admin -v setup`. I got the following output:

```
please run './securedrop-admin setup' first
```
So then I upgraded to 0.6 by doing a `git checkout` to the 0.6 tag. 

I reran `./securedrop-admin -v setup` and got the following stack trace:

```
amnesia@amnesia:~/Persistent/securedrop$ ./securedrop-admin -v setup 
INFO: Virtualenv already exists, not creating
INFO: Checking Python dependencies for securedrop-admin
DEBUG: ERROR: /home/amnesia/Persistent/securedrop/admin/.venv/bin/pip cannot be found.

ERROR: Failed to install pip dependencies. Check network connection and try again.
Traceback (most recent call last):
  File "./admin/bootstrap.py", line 213, in <module>
    args.func(args)
  File "./admin/bootstrap.py", line 149, in envsetup
    install_pip_dependencies(args)
  File "./admin/bootstrap.py", line 186, in install_pip_dependencies
    stderr=subprocess.STDOUT)
  File "/usr/lib/python2.7/subprocess.py", line 219, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['torify', '/home/amnesia/Persistent/securedrop/admin/.venv/bin/pip', 'install', '-r', '/home/amnesia/Persistent/securedrop/admin/requirements.txt', '--require-hashes', '-U', '--upgrade-strategy', 'only-if-needed']' returned non-zero exit status 1
```

Curiously, I again ran `./securedrop-admin setup` and got another error
```
INFO: Virtualenv already exists, not creating
INFO: Check Python dependencies for securedrop-admin
ERROR: failed to install pip dependencies. Check network connections and try again
````

It should be pointed out that networking is not the issue as I am fully connected to Tor browser and can access the web. Running a `rm -rf ~/Persistence/securedrop/admin/.venv/` and rerunning the setup script resolved the issue for me.
 
## Expected Behavior

The setup script should complete without issue. If it does fail, the output should be indicative of a the actual error.

## Actual Behavior

There seems to be some errors in the the upgrade path. At minimum, our documentation doesn't point to the new paths and should be updated.

**Comment by dachary** [comment]
Here is my theory about what happened:

* `rm -f admin/.venv/*` was done, or something similar leaving an empty but existing `admin/.venv` directory
* because `./securedrop-admin setup` tests only the existence of the `admin/.venv` directory, it assumed all is well. And then failed because well, `pip` was nowhere to be found
* when `admin/.venv` was entirely removed, `./securedrop-admin setup` correctly did the setup

Could this be what happened? If not I'm puzzled as to how the `admin/.venv` came to be created empty, as if the `virtualenv` command that creates it was interrupted in the middle.

Side note: I agree with you, the `ERROR: failed to install pip dependencies. Check network connections and try again` makes very little sense because it wrongly assumes **all** errors are because of network connections. It should be modified. Also note that you get the same error message with and without the **-v** flag. Only with the **-v** flag you have additional messages.

**Comment by dachary** [comment]
This has not been reported to be a problem in the past months. It could be because I'm not reading support tickets though. @b-meson @redshiftzero have you heard about this issue in the recent past?

**Comment by redshiftzero** [comment]
I haven't seen this in any support tickets in the last few months - @edenemmanuel @b-meson let me know if it happened and I missed it.

**Comment by b-meson** [comment]
I think this is an edge case but from when I removed the `venv` directory while I was debugging another issue. I think the path at `~/Persistence/securedrop/.venv/` was symlinked to `~/Persistence/securedrop/admin/.venv/`  Removing the directory does not remove the symlink. Isit possible to recreate it because the [docs](https://docs.securedrop.org/en/release-0.6/install.html?highlight=venv) are incomplete. We did make a change to the docs in 48a560b80 but they do not reference the `admin` path. 

There have been no new reports of this bug.  I am fine with closing if it is a release blocker @redshiftzero @dachary
